packages/vitest/src/node/core.ts

```diff
@@ -423,11 +423,30 @@ export class Vitest {
     return join(configDir, workspaceConfigName)
   }
 
-  private async resolveWorkspace(cliOptions: UserConfig): Promise<TestProject[]> {
+  private async resolveProjects(cliOptions: UserConfig): Promise<TestProject[]> {
     const names = new Set<string>()
 
+    if (this.config.projects) {
+      if (typeof this.config.workspace !== 'undefined') {
+        this.logger.warn(
+          'Both `config.projects` and `config.workspace` are defined. Ignoring the `workspace` option.',
+        )
+      }
+
+      return resolveProjects(
+        this,
+        cliOptions,
+        undefined,
+        this.config.projects,
+        names,
+      )
+    }
+
     if (Array.isArray(this.config.workspace)) {
-      return resolveWorkspace(
+      this.logger.deprecate(
+        'The `workspace` option is deprecated and will be removed in the next major. To hide this warning, move `workspace` array to `projects`.',
+      )
```

Comment:
`move` sounds a bit weird, maybe `rename` instead?

```suggestion
      this.logger.deprecate(
        'The `workspace` option is deprecated and will be removed in the next major. To hide this warning, rename `workspace` option to `projects`.',
      )
```